.github/workflows/pr-code-checks.yml

```diff
@@ -103,7 +106,7 @@ jobs:
         run: |
           make -j2 generate
           git diff --exit-code ':!*go.sum' || (echo 'Generated code is out of date, please run "make generate" and commit the changes in this PR.' && exit 1)
-  unittest:
+  unit-tests:
```

Comment:
Do we need `go-cache`.

Reply:
it is not necessary but should provide a nice build time speedup